action_plugins/ios_user_manager.py

```diff
@@ -0,0 +1,109 @@
+# (c) 2018, Ansible by Red Hat, inc
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'network'}
+import re
+import base64
+import hashlib
+
+from ansible.plugins.action import ActionBase
+
+
+class UserManager:
+
+    def __init__(self, new_users, user_config_data):
+        self.__new_users = new_users
+        self.__user_config_data = user_config_data
+
+    @staticmethod
+    def calculate_fingerprint(sshkey):
+        if ' ' in sshkey:
+            keyparts = sshkey.split(' ')
+            keyparts[1] = hashlib.md5(base64.b64decode(keyparts[1])).hexdigest().upper()
+            return ' '.join(keyparts)
+
+        else:
+            return 'ssh-rsa %s' % hashlib.md5(base64.b64decode(sshkey)).hexdigest().upper()
+
+    def _parse_view(self, data):
+        match = re.search(r'view (\S+)', data, re.M)
+        if match:
+            return match.group(1)
+
+    def _parse_sshkey(self, data):
+        match = re.search(r'key-hash (\S+ \S+(?: .+)?)$', data, re.M)
+        if match:
+            return match.group(1)
+
+    def _parse_privilege(self, data):
+        match = re.search(r'privilege (\S+)', data, re.M)
+        if match:
+            return int(match.group(1))
+
+    def generate_existing_users(self):
+        match = re.findall(r'(?:^(?:u|\s{2}u))sername (\S+)', self.__user_config_data, re.M)
+        if not match:
+            return list()
```

Comment:
minor comment - [] is bit more efficient than list() for empty list

Reply:
+1